### ISSUE TITLE
State of v3 documentation not great

### ISSUE THREAD

**Issue by untoldone** [cornerstonejs/cornerstoneTools#906]
Maybe I've completely missed something .. but I feel like the state of documentation for the project is that it is virtually non-usable for someone starting from scratch.

Individual tools have no documentation, it unclear where to find any fully functioning examples, whole categories of features have no mention (synchronizers, stacks?)

It feels like you folks have put a ton of hard work in and based on some of the third party examples I've seen, I'm really impressed with what you've been able to construct ... but I feel like this effort is all for not for a large percentage of your possible users without complete docs.

Is there anything anyone (at the project, or outside the project) can do to help get the documentation for the latest release in order?

**Comment by dannyrb** [comment]
Existing Resources:

- Docs: https://tools.cornerstonejs.org/
- Examples: https://tools.cornerstonejs.org/examples/
- Auto-generated API Explorer: https://tools.cornerstonejs.org/api/ (takes a second to load)

`cornerstone-tools` is maintained by volunteers. The libraries are grouped under the "Cornerstone.js Organization", but members are really the individuals that consistently contribute and help move the libraries forward. Most changes are "funded" by our respective employers as we add features that help us move whatever projects we're working on forward. Documentation tends to lag because most of our employers don't see it as a direct benefit ¯\\_(ツ)_/¯ 

If there is specific documentation or examples that are missing, hard to follow, or incorrect, please feel free to create a more specific issue or mention it here. Often detailed responses to specific issues/questions make their way into our documentation.

You can add your own example or update the documentation by either:

- Adding or updating a file in [`/examples`](https://github.com/cornerstonejs/cornerstoneTools/tree/master/examples)
- Adding or updating the markdown in [`/docs/latest`](https://github.com/cornerstonejs/cornerstoneTools/tree/master/docs/latest)
- Adding JSDoc to anything publicly exposed (updates API docs)

Examples are a simple Jekyll site, docs are generated with GitBook CLI, and API docs are generated with `Docma`. All of these are built, updated, and published when changes are merged to `master`. If you're encountering issues updating any of these, let me know and I can spell out more detailed instructions (which we could then turn into a mini-guide to help other would-be contributors).

**tl;dr**

- Be the change you want to see! (PR some doc updates 🙌)
- OR at least help us understand which areas you're having the most trouble with so we can prioritize our incredibly limited resources
- OR hire one of the consultancies that contribute to this project 😛

**Comment by untoldone** [comment]
Thanks for the detailed response -- understood on the constraints. Again, really impressed with the things I've seen so far, just struggling to get purchase. That said, I'll try to add where I think it might help.

**Comment by untoldone** [comment]
@dannyrb I'm reading through code to figure stuff out at this point, but given the number of abstractions, it is hard for me to get a handle of how these are intended to be used together -- also given it looks like OHIF largely uses v2 / I can't really find a "kitchen sink" example or integration tests in the wild to learn from ... I'm finding I have a number of quick questions to help figure some stuff out but I don't feel a git issue is the appropriate venue.

Is there anywhere where folks congregate (e.g. slack, gitter, irc, stackoverflow etc?)

Thanks!

**Comment by dannyrb** [comment]
No worries. I appreciate criticism, I just want to make sure it's something targeted/actionable so we can use it to improve.

The OHIF Viewer is undergoing a large refactor to migrate away from `meteor` and toward a `react` client. The react version appears to be using `tools@3`. I think the most recent code is here:

- [`react` branch: /Packages-react/ohif-viewer](https://github.com/OHIF/Viewers/tree/react/Packages-react/ohif-viewer)
- [With some updated docs hanging out here](https://deploy-preview-321--ohif.netlify.com/essentials/getting-started.html)

FWIW, I've been told that OHIF has the `react` version's release, and making it easier to contribute/get-started as a priority.

We also have a _very basic_ Viewer implementation in the [`index.html` in our root](https://github.com/cornerstonejs/cornerstoneTools/blob/master/index.html). We use it as a quick regression test on Netlify previews for PRs: https://deploy-preview-905--cornerstone-tools.netlify.com/

If you have well-formed questions, detail what you've already tried, etc. I'm always happy to respond. We're not popular enough for StackOverflow to be a great place to seek answers, so it makes sense that they would filter in here.

> Is there anywhere folks congregate where quick questions are appropriate (e.g. slack, gitter, irc, stackoverflow etc)?

If you shoot your email address over to me @ `danny.ri.brown@gmail.com`, I can invite you to the maintainers Slack group. We have channels for `tools` / `help`.

> Also do you know of any open source projects that use a larger swath of v3 features than the examples today to learn from?

I'm aware of quite a few closed source viewers built with `cornerstone.js`, but OHIF is the only open source one I know. I'm remember hearing of others, but haven't actively searched in quite some time.

**Comment by untoldone** [comment]
Awesome, the Netlify demo is perfect. I wasn't entirely sure what that was when I had seen it first time around / skipped over it.

I had come across the standalone react viewer for OHIF that referenced v3 but it didn't seem like it had much interface to the tools yet -- I hadn't seen the branch, but I'll check that out too.

Will do on the email for slack access.

I'm ok if you want to go ahead and close this unless you want to leave it for others. I'll open more focused documentation / code bugs as I come across them or try to fix them as I can. Thanks for helping me get started here!

**Comment by dannyrb** [comment]
I've created the following issue to address what I believe are the core takeaways from our discussion:

https://github.com/cornerstonejs/cornerstoneTools/issues/909

Thanks again for stopping in and speaking up 🙌

**Comment by untoldone** [comment]
Thanks -- that looks like an appropriate issue given our discussion! I'll probably be in and out of submitting a few things here and there as I get started over the next few weekends -- thanks for helping here!
